frontend/src/components/pages/AddProduct.js

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from "react";
+import ProductDataService from "../../client/product";
+import styled from "styled-components";
+
+const AddProduct = () => {
+    const initialProductState = {
+        title: "",
+        description: "",
+        price: "",
+        stock: "",
+        isActive: true,
+        categoryId: ""
+
+    };
+    const [product, setProduct] = useState(initialProductState);
+    const [submitted, setSubmitted] = useState(false);
+
+    const handleInputChange = event => {
+        const { name, value } = event.target;
+        setProduct({...product, [name]: value});
+    };
+
+
+    const saveProduct = () => {
+        var data = {
+            title: product.title,
+            description: product.description,
+            price: product.price,
+            stock: product.stock,
+            isActive: product.isActive,
+            categoryId: product.categoryId
+        };
+
+        ProductDataService.create(data)
```

Comment:
Como o seu `product` state já tem exatamente essa mesma estrutura, acredito que você poderia pular essa passo a mais que não altera em nada os dados:

```suggestion
    const saveProduct = () => {
        ProductDataService.create(product)
```